Bootcamp2015-AmazingRace/Views/MainPage.xaml.cs

```diff
@@ -30,5 +36,25 @@ protected override void OnNavigatedTo(NavigationEventArgs e)
             // If you are using the NavigationHelper provided by some templates,
             // this event is handled for you.
         }
+
+        private async void GoogleLoginClick(object sender, Windows.UI.Xaml.RoutedEventArgs e)
+        {
+            var result = await App.Api._mobileServiceClient.LoginAsync(MobileServiceAuthenticationProvider.Google);
```

Comment:
Accessing the private field (_mobileServiceClient)? Seems like it would have made more sense to put a method on Api for this. 

This is a good example of Law of Demeter (dot-counting) violation. App DOT Api DOT _mobileServiceClient DOT Login.  If the Api class was passed into the ViewModel via Constructor Injection and the API shielded you from the Mobile service Client. It would be Api.LoginAsync. Much easier to test.